README.md

```diff
@@ -23,11 +23,33 @@ In this challenge, you are to build the Smurfs village once again, only this tim
 Demonstrate your understanding of this Sprint's concepts by answering the following free-form questions. Edit this document to include your answers after each question. Make sure to leave a blank line above and below your answer so it is clear and easy to read by your project manager.
 
 - [ ] What problem does the context API help solve?
+
+We use the Context API when we have global data that lots of components share (user, theme, etc), or when we have to pass data through intermediate components. This helps us keep our state relatively clean.
+
+
+
 - [ ] In your own words, describe `actions`, `reducers` and the `store` and their role in Redux. What does each piece do? Why is the store known as a 'single source of truth' in a redux application?
+
+actions => Are events that we define that happen which lead to the next state.
+reducers => It provides us with the previous states along with the enxt state.
+store => It contains the state for our app.
+
+
+
 - [ ] What is the difference between Application state and Component state? When would be a good time to use one over the other?
+
+Application State is immutable while component state is mutable. 
+
+
 - [ ] Describe `redux-thunk`, what does it allow us to do? How does it change our `action-creators`?
+
+Redux Thunk is a middleware created by Dan Abramov, that provides the ability to handle asynchronous operations inside our Action Creators. It changes our action-creators as now we can make API calls from our action creators.
+
+
 - [ ] What is your favorite state management system you've learned and this sprint? Please explain why!
 
+  Redux, it is the easiest to use and feels like it has the most functionality. 
```

Comment:
ayeeeeee, it feels like the most functionality cause its the most built out, as its the oldest solution to dealing with state available for react i believe